### PR TITLE
Abstract the creation component

### LINKED ISSUE #17: Add create component to BaseActivity

### PR DESCRIPTION
**Pull request by Numan1617**
Closes #17.

### CODE REVIEW

File: app/src/main/java/com/ustwo/boilerplate/base/BaseActivity.java [hunk 1]
```diff
@@ -29,4 +30,6 @@ protected void onDestroy() {
 
   @NonNull
   protected abstract V getPresenterView();
+
+  protected abstract C createComponentAndInject();
```
Comment (jarekankowski) [inline app/src/main/java/com/ustwo/boilerplate/base/BaseActivity.java:41]: I think you can call createComponentAndInject() method from onCreate() and get rid of it from the children of the activity

File: app/src/main/java/com/ustwo/boilerplate/feature/helloworld/HelloWorldActivity.java [hunk 2]
```diff
@@ -36,7 +36,8 @@ protected int getLayoutId() {
     return this;
   }
 
-  private HelloWorldComponent createComponentAndInject() {
+  @Override
+  protected HelloWorldComponent createComponentAndInject() {
```
Comment (jarekankowski) [inline app/src/main/java/com/ustwo/boilerplate/feature/helloworld/HelloWorldActivity.java:32]: You return component and don't use it. Consider changing method to return void.

Comment (jarekankowski) [reply app/src/main/java/com/ustwo/boilerplate/feature/helloworld/HelloWorldActivity.java:32]: As we need ApplicationComponent in every activity maybe it'll be useful to create method in BaseActivity to return it?

**Review by jarekankowski — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/ustwo/boilerplate/feature/helloworld/HelloWorldActivity.java (hunk 2 above)
Comment (Numan1617) [reply app/src/main/java/com/ustwo/boilerplate/feature/helloworld/HelloWorldActivity.java:32]: Left it open to ease extension from tests or sub classes

File: app/src/main/java/com/ustwo/boilerplate/feature/helloworld/HelloWorldActivity.java [hunk 3]
```diff
@@ -3,19 +3,17 @@
...
 import javax.inject.Inject;
 
-public class HelloWorldActivity extends BaseActivity<HelloWorldPresenter.View>
+public class HelloWorldActivity extends BaseActivity<HelloWorldPresenter.View, HelloWorldComponent>
     implements HelloWorldPresenter.View {
 
   @Inject HelloWorldPresenter presenter;
 
   @Override
   protected void onCreate(final Bundle savedInstanceState) {
```
Comment (jarekankowski) [inline app/src/main/java/com/ustwo/boilerplate/feature/helloworld/HelloWorldActivity.java]: Really useful method

**Review by jarekankowski — CHANGES_REQUESTED** [review]

**Review by jarekankowski — APPROVED** [review]

**Pull request merged**